addons/events/fnc_weaponEvents.sqf

```diff
@@ -0,0 +1,80 @@
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_weaponEvents
+
+Description:
+    Execute weapon events framework.
+
+    class MyWeapon: MyWeapon_base {
+        class EventHandlers {
+            fired = "_this call CBA_fnc_weaponEvents";
+        };
+    };
+
+Parameters:
+    _unit   - Unit that fired the weapon <OBJECT>
+    _weapon - The weapon fired by the unit <STRING>
+
+Returns:
+    Nothing
+
+Examples:
+    (begin example)
+        fired = "_this call CBA_fnc_weaponEvents";
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+
+params ["_unit", "_weapon"];
+
+private _config = configFile >> "CfgWeapons" >> _weapon >> "CBA_WeaponEvents";
+
+private _isEmpty = _unit ammo _weapon == 0;
+private _onEmpty = true;
+
+if (isNumber (_config >> "onEmpty")) then {
+    _onEmpty = getNumber (_config >> "onEmpty") == 1;
```

Comment:
default returns 0, so can probably skip `isNumber`?

Reply:
But I want 1/true to be the default.

Reply:
why not `CBA_fnc_getConfigEntry`?

Reply:
Overhead.